build/org.eclipse.cdt.core.autotools.core/src/org/eclipse/cdt/core/autotools/core/AutotoolsBuildConfiguration.java

```diff
@@ -65,7 +63,8 @@ public IProject[] build(int kind, Map<String, String> args, IConsole console, IP
 
 		IProject project = getProject();
 
-		execute(Arrays.asList(new String[] { "autoreconf", "--install" }), project.getLocation(), console, monitor); //$NON-NLS-1$ //$NON-NLS-2$
+		executeRemote(Arrays.asList(new String[] { "autoreconf", "--install" }), project.getLocation(), console, //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
Oh wait, when I try your change locally I get this in the build log:

```
Building in: /home/mbooth/runtime-cdt-0/AutoTest/build/default
autoreconf: error: 'configure.ac' is required
```

I believe "executeRemote" uses the wrong working directory for "autoreconf" to work ---- the "execute" method used `project.getLocation()` as the working dir, but "executeRemote" uses `getBuildDirectory()` which evaluates to something like `project.getLocation() + /build/default`

Consider passing the desired working directory into the "executeRemote" method

Reply:
I hoisted the working directory out of the execute function so that the caller can decide the correct path and fixed some related code along the way (like error handling and redundant code)